plugins/sentinelone/komand_sentinelone/util/api.py

```diff
@@ -47,6 +46,39 @@ def clean(obj):
     return cleaned
 
 
+def set_agents_array(search: str, agent_details: str, agents: list) -> list:
+    """
+    Checks Search and assigns agents appropriate values
+    :param search: String that will be searched for
+    :param agent_details: Details of agent
+    :param agents: List of agents
+    :returns agents: List of agents
+    """
+    # Normalize casing if specified
+    if search == "computerName":
+        agents = [agent_details.lower(), agent_details.upper()]
+    if search == "uuid":
+        agents = [agent_details.lower()]
+    return agents
+
+
+def get_agents_data(self, agent: str, api_version: str, search: str, results: list):
+    """
+    Gets agents Data
+    :param self: Self
+    :param agent: Agent to get details for
+    :param api_version: Version of API
+    :param search: String that will be searched for
+    :param results: Resulting list of agent details
+    """
+    endpoint = f"{self.url}web/api/v{api_version}/agents?{search}={agent}"
+    output = requests.get(endpoint, headers=self.token_header)
+    if output.status_code == 200 and output.json().get("pagination", {}).get("totalItems", 0) >= 1:
+        agents_data = output.json().get("data", [])
+        if agents_data and agents_data[0] not in results:
+            results.append(agents_data[0])
+
+
```

Comment:
Why those methods are outside the `SentineloneAPI` class? Kindly, please move them there 😄